src/dependency-manager/src/service-config/builder.ts

```diff
@@ -13,31 +15,68 @@ class MissingConfigFileError extends Error {
 }
 
 export class ServiceConfigBuilder {
-  static CONFIG_FILENAME = 'architect.json';
+  static getConfigPaths(input: string) {
+    return [
+      input,
+      path.join(input, 'architect.json'),
+      path.join(input, 'architect.yml'),
+      path.join(input, 'architect.yaml'),
+    ];
+  }
+
+  static buildFromPath(input: string): ServiceConfig {
+    const try_files = ServiceConfigBuilder.getConfigPaths(input);
 
-  static buildFromPath(service_path: string): ServiceConfig {
-    let config_path = service_path;
-    if (!service_path.endsWith('.json')) {
-      config_path = path.join(service_path, ServiceConfigBuilder.CONFIG_FILENAME);
+    // Make sure the file exists
+    let file_contents;
+    for (const file of try_files) {
```

Comment:
You also need to update the local dependency manager. Supports explicit architect.json and now yaml ref.
```
const node = new LocalServiceNode({
      service_path: service_path.endsWith('.json') ? path.dirname(service_path) : service_path,
      service_config: service_config,
      node_config: this.getNodeConfig(service_config, 'latest'),
      image: service_config.getImage(),
      tag: 'latest',
    });
```